Cadastro de usuário técnico

luc527 opened the pull request.

luc527 committed:
cadastro feito; início telas de login

luc527 committed:
cont. tela login

luc527 committed:
login implementado

adicionada dependência ao VarDumper do symfony, gera uma interface bem
melhor que o var_dump do PHP

usar com a função dump($x)

luc527 committed:
add tests for App\Tecnico\Clube

luc527 committed:
cadastos e logins refatorados com classe SenhaCriptografada

login de técnico e admin refatorado:
- classes Login e LoginRepository removidas, funcionalidade de criptografia das senhas fica na classe SenhaCriptografada agora (removidos testes)
- criado Admin e AdminRepository
- login e cadastro de admin e técnico alterados para usar a SenhaCriptofada

melhorias menores:
- para não ficar repetindo data alteração / criação, criados traits

luc527 committed:
refatorar Competicao, agora tem datas de criação e alteração no php

luc527 committed:
fix: bug no CompeticaoRepositoryTest após adicionar data de criação e alteração

ghost commented:
![Failed](http://localhost:9000/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- D Security Rating on New Code (is worse than A)
- 13.30% Coverage on New Code (is less than 80.00%)
- 0.00% Security Hotspots Reviewed on New Code (is less than 100.00%)

# Analysis Details
## 15 Issues
- ![Bug](http://localhost:9000/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](http://localhost:9000/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 2 Vulnerabilities
- ![Code Smell](http://localhost:9000/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 13 Code Smells

## Coverage and Duplications
- ![Coverage](http://localhost:9000/static/communityBranchPlugin/checks/CoverageChart/0.svg?sanitize=true) 13.30% Coverage (31.50% Estimated after merge)
- ![Duplications](http://localhost:9000/static/communityBranchPlugin/checks/Duplications/3.svg?sanitize=true) 0.00% Duplicated Code (0.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](http://localhost:9000/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=19)

luc527 commented:
Logo resolvo os problemas do SonarQube (code smells, testes etc)

luc527 committed:
classe helper Dates

para não escrever DateTimeImmutable::createFromFormat(...) e
->format('Y-m-d H:i:s.u') o tempo todo

ghost commented:
![Failed](http://localhost:9000/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- D Security Rating on New Code (is worse than A)
- 13.20% Coverage on New Code (is less than 80.00%)
- 0.00% Security Hotspots Reviewed on New Code (is less than 100.00%)

# Analysis Details
## 9 Issues
- ![Bug](http://localhost:9000/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](http://localhost:9000/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 2 Vulnerabilities
- ![Code Smell](http://localhost:9000/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 7 Code Smells

## Coverage and Duplications
- ![Coverage](http://localhost:9000/static/communityBranchPlugin/checks/CoverageChart/0.svg?sanitize=true) 13.20% Coverage (31.30% Estimated after merge)
- ![Duplications](http://localhost:9000/static/communityBranchPlugin/checks/Duplications/3.svg?sanitize=true) 0.00% Duplicated Code (0.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](http://localhost:9000/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=19)

luc527 committed:
fix: Dates::parseXxxx methods didn't parse nulls correctly

luc527 committed:
Ação de "realizar login" refatorada para testabilidade

Foi criada uma classe "RealizarLogin" responsável somente pela lógica da realização de login usando inversão de controle (ela recebe instâncias do repositório e da sessão no construtor em vez de criar um TecnicoRepository ou usar a $_SESSION). Assim, podemos criar um mock para o TecnicoRepository e uma instância da Session que na verdade usa um array definido pelo programador e testar sem causar efeitos colaterais ou depender do banco de dados. *Acho* que os "use case" da clena architecture são assim.

Para ela também não ter que lidar com a validação da request (verificar que existem campos 'email' e 'senha', sanitizar etc.), foi criado um LoginDTO que representa os dados da request já validados. Esse é o objeto passado ao RealizarLogin.

Para controlarmos os dados em sessão, criamos uma classe Session que, em produção, vai encapsular o $_SESSION, mas nos testes pode encapsular qualquer array. A classe Session de antes foi renomeada para SessionOld e os seus usos serão atualizados para usar a Session nova.

Dessa forma, a ação no login/acao.php só é responsável por parsear a request num LoginDTO, criar o repositório, encapsular a sessão no objeto Session e invocar o RealizarLogin -- nenhuma lógica de negócio.

Também já foi feito um RealizarLoginTest para demonstrar a testabilidade dessa abordagem.

ghost commented:
![Failed](http://localhost:9000/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- D Security Rating on New Code (is worse than A)
- 45.80% Coverage on New Code (is less than 80.00%)
- 0.00% Security Hotspots Reviewed on New Code (is less than 100.00%)

# Analysis Details
## 13 Issues
- ![Bug](http://localhost:9000/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](http://localhost:9000/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 2 Vulnerabilities
- ![Code Smell](http://localhost:9000/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 11 Code Smells

## Coverage and Duplications
- ![Coverage](http://localhost:9000/static/communityBranchPlugin/checks/CoverageChart/40.svg?sanitize=true) 45.80% Coverage (55.30% Estimated after merge)
- ![Duplications](http://localhost:9000/static/communityBranchPlugin/checks/Duplications/3.svg?sanitize=true) 0.00% Duplicated Code (0.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](http://localhost:9000/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=19)

MateuxLucax committed:
Merge branch 'main' into cadastro-tecnico

# Conflicts:
#	public/admin/competicoes/acao.php
#	public/admin/competicoes/index.php
#	src/Admin/Competicoes/CompeticaoRepository.php
#	tests/Admin/Competicoes/CompeticaoRepositoryTest.php

MateuxLucax committed:
refac + merge: merge main on branch + init refac to comply w/ PSR

MateuxLucax committed:
test: fix tests by removing readonly from TecnicoRepository

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- D Security Rating on New Code (is worse than A)
- 45.70% Coverage on New Code (is less than 80.00%)
- 0.00% Security Hotspots Reviewed on New Code (is less than 100.00%)

# Analysis Details
## 8 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 2 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 6 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/40.svg?sanitize=true) 45.70% Coverage (56.10% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/3.svg?sanitize=true) 0.00% Duplicated Code (0.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=19)

MateuxLucax committed:
fix: replace rand with random_int (security)

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- D Security Rating on New Code (is worse than A)
- 45.70% Coverage on New Code (is less than 80.00%)

# Analysis Details
## 7 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 1 Vulnerability
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 6 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/40.svg?sanitize=true) 45.70% Coverage (56.10% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/3.svg?sanitize=true) 0.00% Duplicated Code (0.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=19)

MateuxLucax committed:
refac: finish SonarLint comply

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- 40.80% Coverage on New Code (is less than 80.00%)

# Analysis Details
## 0 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 0 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/40.svg?sanitize=true) 40.80% Coverage (54.80% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/3.svg?sanitize=true) 0.00% Duplicated Code (0.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=19)

luc527 committed:
classe Cadastrar + simplificação dos erros

os erros eram criados de algumas formas diferentes:

- tinha a classe Result que tinha a desvantagem de perder o tipo dos
objetos (atributo $data era mixed)

- os DTOs retornavam DTO|string

- e os casos de uso (realizar login, cadastrar) faziam throw de
ValidatorException mas também davam catch nelas e transformavam em
Result

agora é feito tudo por exception
quando é uma exception que queremos retornar para o usuário não com
código 500 e mostrando uma mensagem específica ao invés de "ocorreu um
erro inesperado", se usa a ValidatorException

assim a maioria do código não lida com as exceções, isso fica só lá onde
o controller é chamado (catch Exception $e -> return Response::erroException($e))

os testes também foram atualizados

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- 47.80% Coverage on New Code (is less than 80.00%)

# Analysis Details
## 7 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 7 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/40.svg?sanitize=true) 47.80% Coverage (58.10% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/3.svg?sanitize=true) 0.00% Duplicated Code (0.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=19)

MateuxLucax committed:
refac: remove ResponseException. Migrate to ValidatorException

luc527 committed:
fix bugs in test cases

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- 60.70% Coverage on New Code (is less than 80.00%)

# Analysis Details
## 4 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 4 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/60.svg?sanitize=true) 60.70% Coverage (68.70% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/3.svg?sanitize=true) 0.00% Duplicated Code (0.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=19)

MateuxLucax committed:
test: add missing tests

MateuxLucax committed:
test: fix TecnicoRepositoryArray tests

ghost commented:
![Passed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/passed.svg?sanitize=true)

# Analysis Details
## 0 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 0 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/90.svg?sanitize=true) 96.90% Coverage (97.50% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/3.svg?sanitize=true) 0.00% Duplicated Code (0.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=19)

MateuxLucax committed:
test: adicionado testes para finalizar o coverage

ghost commented:
![Passed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/passed.svg?sanitize=true)

# Analysis Details
## 0 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 0 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/100.svg?sanitize=true) 100.00% Coverage (100.00% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/3.svg?sanitize=true) 0.00% Duplicated Code (0.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=19)
